core/src/main/java/org/elasticsearch/common/network/NetworkService.java

```diff
@@ -106,31 +117,44 @@ public void addCustomNameResolver(CustomNameResolver customNameResolver) {
             }
         }
         // finally, fill with our default
-        if (bindHost == null) {
-            bindHost = DEFAULT_NETWORK_HOST;
+        if (bindHosts == null) {
+            bindHosts = new String[] { DEFAULT_NETWORK_HOST };
```

Comment:
our default is a single host specification: `_local_`. One point of this change is that you can use multiple: e.g. `_local_,12.1.4.2,foo.bar.com`